### PR TITLE
PHANTASM recipe

### PR DESCRIPTION
**Pull request by dr-joe-wirth**
Adding a recipe for `phantasm`. Currently build works and tests are passing. Still want to do a little polishing before publishing. Requesting @rpetit3 as a reviewer.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by dr-joe-wirth** [comment]
@BiocondaBot please fetch artifacts

**Comment by dr-joe-wirth** [comment]
@rpetit3 not sure why the artifacts thing failed.

**Comment by rpetit3** [comment]
@BiocondaBot please fetch artifacts

**Comment by rpetit3** [comment]
@dr-joe-wirth the ci tests mighr have still been running when you issued the first command

**Comment by dr-joe-wirth** [comment]
@BiocondaBot please fetch artifacts

**Comment by rpetit3** [comment]
Once green check mark is there for the latest commit, then the artifacts will be available

**Comment by dr-joe-wirth** [comment]
yeah i figured. it should be working now, but I want to make sure everything runs properly before giving the green light to merge. i will comment here again once i have confirmed things look good.

**Comment by dr-joe-wirth** [comment]
@BiocondaBot please fetch artifacts

File: recipes/phantasm/build.sh
Comment (bgruening) [inline recipes/phantasm/build.sh:42]: Why can you not use the prefix/bin this would be already on Path and would simplify this build script a lot.

**Comment by dr-joe-wirth** [comment]
@BiocondaBot please fetch artifacts

File: recipes/phantasm/build.sh
Comment (dr-joe-wirth) [reply recipes/phantasm/build.sh:42]: I prefer to keep all the source code in its own folder instead of comingling with everything in `bin`. It seems a lot more organized to me and only requires a few extra commands. It also prevents any unexpected collisions. Like if another package has a python file named `main.py`. PHANTASM will break if that file gets clobbered.

Comment (bgruening) [reply recipes/phantasm/build.sh:42]: But because of that you need to change the PATH via an activate script, which is also very ugly and does not work well with containers. Have you checked if the containers are working?

Comment (dr-joe-wirth) [reply recipes/phantasm/build.sh:42]: the containers are working. why is it ugly? i am new to `conda` recipes.

Comment (dr-joe-wirth) [reply recipes/phantasm/build.sh:42]: i'd be happy to plop the `phantasm.py` file into `$PREFIX/bin` but idk how to do that and still have it reference all the files it needs to import in the `phantasm` folder.

**Comment by dr-joe-wirth** [comment]
@BiocondaBot please fetch artifacts

**Comment by dr-joe-wirth** [comment]
@rpetit3 this recipe is working. It is ready to merge unless you want me to address the style issues that @bgruening brought up. Their main concern was that I am modifying the user's `PATH` with a activate script. I could avoid this if I knew how to place a file named `phantasm.py` inside the `$PREFIX/bin` folder so that it can still import the code from within the `phantasm` directory. Unfortunately, I have no idea how to do this, but I'd be happy to try to figure it out if you think that would be a preferrable solution.

**Comment by dr-joe-wirth** [comment]
@rpetit3 actually, give me until next week to make my recipe less "ugly". I think I can do something a little more elegant to avoid modifying the user's `PATH`.

**Comment by dr-joe-wirth** [comment]
@rpetit3 this recipe is working and ready to merge.

**Comment by dr-joe-wirth** [comment]
@BiocondaBot please fetch artifacts

**Comment by dr-joe-wirth** [comment]
@rpetit3 this recipe is working and ready to be merged.

**Review by rpetit3 — APPROVED** [review]

**Pull request merged**
